client/pages/venue.jsx

```diff
@@ -10,21 +12,34 @@ const Venue = () => {
 
     menuClass = 'show';
   }
+
   const onSave = e => {
-    alert('save');
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(venue)
+    };
+    fetch('/api/user/addFavorite/', requestOptions).then(res => {
+      alert(res);
+    });
   };
   const onMenu = e => {
     window.open(venue.menuUrl);
   };
   const mainPageClicked = e => {
+    localStorage.removeItem('favorites');
+    window.location.replace('/');
+  };
+  const favorites = e => {
+    localStorage.setItem('favorites', true);
```

Comment:
If you are using `react-router`, then you should use their `<Link />` component instead of onClick.